homeassistant/components/enasolar/sensor.py

```diff
@@ -0,0 +1,248 @@
+"""EnaSolar solar inverter interface."""
+
+from __future__ import annotations
+
+from datetime import date, datetime
+import logging
+
+from homeassistant.components.sensor import (
+    STATE_CLASS_MEASUREMENT,
+    STATE_CLASS_TOTAL_INCREASING,
+    SensorEntity,
+)
+from homeassistant.const import (
+    CONF_NAME,
+    DEVICE_CLASS_ENERGY,
+    DEVICE_CLASS_POWER,
+    DEVICE_CLASS_TEMPERATURE,
+    ENERGY_KILO_WATT_HOUR,
+    EVENT_HOMEASSISTANT_START,
+    POWER_KILO_WATT,
+    TEMP_CELSIUS,
+    TEMP_FAHRENHEIT,
+)
+from homeassistant.core import CALLBACK_TYPE, callback
+from homeassistant.helpers.event import async_call_later
+from homeassistant.util import dt as dt_util
+
+from .const import (
+    CONF_CAPABILITY,
+    CONF_DC_STRINGS,
+    CONF_MAX_OUTPUT,
+    CONF_SUN_DOWN,
+    CONF_SUN_UP,
+    DEFAULT_SUN_DOWN,
+    DEFAULT_SUN_UP,
+    DOMAIN,
+    ENASOLAR,
+    ENASOLAR_UNIT_MAPPINGS,
+    SCAN_DATA_MIN_INTERVAL,
+    SCAN_MAX_INTERVAL,
+    SCAN_METERS_MIN_INTERVAL,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add enasolar entry."""
+
+    # Use all sensors by default, but split them to have two update frequencies
+    hass_meter_sensors = []
+    hass_data_sensors = []
+
+    enasolar = hass.data[DOMAIN][config_entry.entry_id][ENASOLAR]
+
+    if config_entry.options != {}:
+        enasolar.sun_up = dt_util.parse_time(config_entry.options[CONF_SUN_UP])
+        enasolar.sun_down = dt_util.parse_time(config_entry.options[CONF_SUN_DOWN])
+    else:
+        enasolar.sun_up = dt_util.parse_time(DEFAULT_SUN_UP)
+        enasolar.sun_down = dt_util.parse_time(DEFAULT_SUN_DOWN)
+
+    enasolar.capability = config_entry.data[CONF_CAPABILITY]
+    enasolar.dc_strings = config_entry.data[CONF_DC_STRINGS]
+    enasolar.max_output = config_entry.data[CONF_MAX_OUTPUT]
+
+    _LOGGER.debug("   Polling between %s and %s", enasolar.sun_up, enasolar.sun_down)
+    _LOGGER.debug(
+        "   Max Output: %s, DC Strings: %s, Capability: %s",
```

Comment:
Is there a reason for prefixing debug messages with whitespace?

Reply:
Just personal preference, I wanted the block of debug messages to standout while I was debugging

Reply:
CI will complain so get it fixed

Reply:
Have removed